packages/caip/src/caip2/caip2.ts

```diff
@@ -32,72 +32,84 @@ type ToCAIP2Args = {
-      reference: {
-        [NetworkTypes.COSMOSHUB_MAINNET]: ChainReference.CosmosHubMainnet,
-        [NetworkTypes.COSMOSHUB_VEGA]: ChainReference.CosmosHubVega,
-
-        [NetworkTypes.OSMOSIS_MAINNET]: ChainReference.OsmosisMainnet,
-        [NetworkTypes.OSMOSIS_TESTNET]: ChainReference.OsmosisTestnet
-      }
+const shapeShiftToCAIP2 = Object.freeze({
+  [ChainTypes.Ethereum]: {
+    namespace: ChainNamespace.Ethereum,
+    reference: {
+      [NetworkTypes.MAINNET]: ChainReference.EthereumMainnet,
+      [NetworkTypes.ETH_ROPSTEN]: ChainReference.EthereumRopsten,
+      [NetworkTypes.ETH_RINKEBY]: ChainReference.EthereumRinkeby
+    }
+  },
+  [ChainTypes.Bitcoin]: {
+    namespace: ChainNamespace.Bitcoin,
+    reference: {
+      [NetworkTypes.MAINNET]: ChainReference.BitcoinMainnet,
+      [NetworkTypes.TESTNET]: ChainReference.BitcoinTestnet
+    }
+  },
+  [ChainTypes.Cosmos]: {
+    namespace: ChainNamespace.Cosmos,
+    reference: {
+      [NetworkTypes.COSMOSHUB_MAINNET]: ChainReference.CosmosHubMainnet,
+      [NetworkTypes.COSMOSHUB_VEGA]: ChainReference.CosmosHubVega
     }
-  } as const
+  },
+  [ChainTypes.Osmosis]: {
+    namespace: ChainNamespace.Cosmos,
+    reference: {
+      [NetworkTypes.OSMOSIS_MAINNET]: ChainReference.OsmosisMainnet,
+      [NetworkTypes.OSMOSIS_TESTNET]: ChainReference.OsmosisTestnet
+    }
+  }
+})
 
+export const toCAIP2: ToCAIP2 = ({ chain, network }): string => {
   const namespace: ChainNamespace = shapeShiftToCAIP2[chain].namespace
 
   switch (chain) {
-    case ChainTypes.Cosmos: {
+    case ChainTypes.Ethereum: {
       const referenceMap = shapeShiftToCAIP2[chain].reference
       switch (network) {
-        case NetworkTypes.COSMOSHUB_MAINNET:
-        case NetworkTypes.COSMOSHUB_VEGA:
-        case NetworkTypes.OSMOSIS_MAINNET:
-        case NetworkTypes.OSMOSIS_TESTNET: {
+        case NetworkTypes.MAINNET:
+        case NetworkTypes.ETH_ROPSTEN:
+        case NetworkTypes.ETH_RINKEBY: {
           const reference: ChainReference = referenceMap[network]
           const caip2 = `${namespace}:${reference}`
           return caip2
         }
       }
       break
     }
-
-    case ChainTypes.Ethereum: {
+    case ChainTypes.Bitcoin: {
       const referenceMap = shapeShiftToCAIP2[chain].reference
       switch (network) {
         case NetworkTypes.MAINNET:
-        case NetworkTypes.ETH_ROPSTEN:
-        case NetworkTypes.ETH_RINKEBY: {
+        case NetworkTypes.TESTNET: {
           const reference: ChainReference = referenceMap[network]
           const caip2 = `${namespace}:${reference}`
           return caip2
         }
       }
       break
     }
-
-    case ChainTypes.Bitcoin: {
+    case ChainTypes.Cosmos: {
       const referenceMap = shapeShiftToCAIP2[chain].reference
       switch (network) {
-        case NetworkTypes.MAINNET:
-        case NetworkTypes.TESTNET: {
+        case NetworkTypes.COSMOSHUB_MAINNET:
+        case NetworkTypes.COSMOSHUB_VEGA: {
```

Comment:
Not sure how testnet upgrades go, but the same naming comment could be applied here as well. Mainnet currently is generic whereas testnet is explicit in this case. I think explicit is the way to go similar to `ETH_ROPSTEN` vs `ETH_RINKEBY`

Reply:
Are you making a suggested change? If so can you be more clear about what change you want to see? I don't understand what you're trying to say I should change.

Reply:
This is just an additional comment in line with the previous question surrounding naming convention for our network types. I think it deserves a quick discussion over